fs/span-manager/span_manager.go

```diff
@@ -442,16 +380,73 @@ func (m *SpanManager) resolveSpanFromCache(s *span, offsetStart, size compressio
 	return nil, ErrSpanNotAvailable
 }
 
-func (m *SpanManager) fetchSpan(buf []byte, spanID compression.SpanID) error {
+// fetchAndCacheSpan fetches a span, uncompresses it and caches the uncompressed
+// span content.
+func (m *SpanManager) fetchAndCacheSpan(spanID compression.SpanID) ([]byte, error) {
+	// fetch compressed span
+	compressedBuf, err := m.fetchSpanWithRetries(spanID)
+	if err != nil && err != io.EOF {
```

Comment:
what should happen in the case of `err == io.EOF`?

Reply:
From the `ReadAt` doc (https://pkg.go.dev/io#ReaderAt only place where `err` might returned within `fetchSpanWithRetries`).

> If the n = len(p) bytes returned by ReadAt are at the end of the input source, ReadAt may return either err == EOF or err == nil.

This may indicate that we should check `if err != nil && err != io.EOF` in all places where `ReadAt` is called. Meanwhile, we should move the `err != io.EOF` into `fetchSpanWithRetries` when `ReadAt` returns.

Reply:
@sbuckfelder added `EOF` check to other `ReadAt` calls (in a separate commit https://github.com/awslabs/soci-snapshotter/pull/444/commits/b4022ce41036db1711e8431e89632d6b3aa8ae89). PTAL.